src/libical/icalrestriction.c.in

```diff
@@ -99,109 +99,93 @@ int icalrestriction_compare(icalrestriction_kind restr, int count)
 
 /* Special case routines */
 
-static const char *icalrestriction_may_be_draft_final_canceled(
+static const char *icalrestriction_validate_status_value(
     const icalrestriction_record *rec, icalcomponent *comp, icalproperty *prop)
 {
-    icalproperty_status stat = icalproperty_get_status(prop);
-
-    _unused(rec);
-    _unused(comp);
+    if (!prop) return 0;
 
-    if (!(stat == ICAL_STATUS_DRAFT ||
-          stat == ICAL_STATUS_FINAL ||
-          stat == ICAL_STATUS_CANCELLED)) {
-        return
-            "Failed iTIP restrictions for STATUS property. "
-            "Value must be one of DRAFT, FINAL, or CANCELED";
-    }
-
-    return 0;
-}
-
-static const char *icalrestriction_may_be_comp_need_process(const icalrestriction_record *
-                                                            rec, icalcomponent *comp,
-                                                            icalproperty *prop)
-{
     icalproperty_status stat = icalproperty_get_status(prop);
```

Comment:
we need to have the` icalproperty_status stat` declaration above the `if (!prop) return 0;` else ancient C compilers complain